test/functional/ui/embed_spec.lua

```diff
@@ -142,20 +144,9 @@ describe('--embed --listen UI', function()
     helpers.skip(helpers.is_os('win'))
     clear()
     local child_server = assert(helpers.new_pipename())
-
-    local screen = thelpers.screen_setup(0,
-      string.format(
-        [=[["%s", "--embed", "--listen", "%s", "-u", "NONE", "-i", "NONE", "--cmd", "%s"]]=],
-        helpers.nvim_prog, child_server, helpers.nvim_set))
-    screen:expect{grid=[[
-      {1: }                                                 |
-                                                        |
-                                                        |
-                                                        |
-                                                        |
-                                                        |
-      {3:-- TERMINAL --}                                    |
-    ]]}
+    uv.fs_unlink(child_server)
```

Comment:
should we do this in `new_pipename`?

Reply:
I think it's better to let the caller decide whether to remove to pipe.

Reply:
Oh, if the pipe exists and the server fails to start, trying to connect to the pipe will hang. Maybe it's better to remove it then.